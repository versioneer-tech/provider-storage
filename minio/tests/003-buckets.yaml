# Test if XR is not ready since the request has not been granted yet
# Test if XR creates expected resource manifests (buckets, policies, user)
apiVersion: epca.eo/v1beta1
kind: XStorage
metadata:
  labels:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
  name: alice-003
spec:
  owner: alice-003
  buckets:
  - bucketName: alice-003
    discoverable: false
  - bucketName: alice-shared-003
    discoverable: true
  bucketAccessRequests:
  - bucketName: bob-shared-003
    permission: ReadWrite
