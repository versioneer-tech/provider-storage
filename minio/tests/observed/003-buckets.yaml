---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-003-alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-003-alice-shared-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-shared-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: minio.crossplane.io/v1
kind: User
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
