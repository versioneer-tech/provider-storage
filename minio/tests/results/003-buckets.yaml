---
apiVersion: epca.eo/v1beta1
kind: XStorage
metadata:
  labels:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
  name: alice-003
spec:
  bucketAccessRequests:
  - bucketName: bob-shared-003
    permission: ReadWrite
  buckets:
  - bucketName: alice-003
    discoverable: false
  - bucketName: alice-shared-003
    discoverable: true
  owner: alice-003
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: check-policy-alice-003.readwrite.bob-shared-003'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-003-alice-003
    xstorages.epca.eo/discoverable: "false"
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-003-alice-shared-003
    xstorages.epca.eo/discoverable: "true"
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  providerConfigRef:
    name: storage-minio
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: check-policy-alice-003.readwrite.bob-shared-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  name: policy-observer-alice-003.readwrite.bob-shared-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: alice-003.readwrite.bob-shared-003
  managementPolicies:
  - Observe
  providerConfigRef:
    name: storage-kubernetes
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    rawPolicy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::alice-003",
              "arn:aws:s3:::alice-003/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-shared-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    rawPolicy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::alice-shared-003",
              "arn:aws:s3:::alice-shared-003/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-minio
---
apiVersion: minio.crossplane.io/v1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: epca.eo/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    policies:
    - alice-003.owner.alice-003
    - alice-003.owner.alice-shared-003
    userName: alice-003
  providerConfigRef:
    name: storage-minio
  writeConnectionSecretToRef:
    name: alice-003
    namespace: default
