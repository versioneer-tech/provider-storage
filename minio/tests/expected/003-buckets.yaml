---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jane
spec:
  bucketAccessRequests:
  - bucketName: b-john
    reason: Need WriteOnly access
  buckets: []
  principal: jane
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: observe-jane-restricted-b-john, user-jane'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-jane-restricted-b-john
  labels:
    crossplane.io/composite: s-jane
  name: observe-jane-restricted-b-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: Policy
      metadata:
        name: jane.restricted.b-john
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jane
  labels:
    crossplane.io/composite: s-jane
  name: user-jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: minio.crossplane.io/v1
      kind: User
      metadata:
        name: jane
      spec:
        forProvider:
          policies: []
          userName: jane
        providerConfigRef:
          name: provider-minio
        writeConnectionSecretToRef:
          name: jane
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
